src/mixer/basetrackplayer.cpp

```diff
@@ -265,11 +265,13 @@ void BaseTrackPlayerImpl::loadTrack(TrackPointer pTrack) {
         }
 
         if (pLoopCue) {
-            double loopStart = pLoopCue->getPosition().toEngineSamplePosMaybeInvalid();
-            double loopEnd = loopStart + (pLoopCue->getLengthFrames() * mixxx::kEngineChannelCount);
-            if (loopStart != kNoTrigger && loopEnd != kNoTrigger && loopStart <= loopEnd) {
-                m_pLoopInPoint->set(loopStart);
-                m_pLoopOutPoint->set(loopEnd);
+            const mixxx::audio::FramePos loopStart = pLoopCue->getPosition();
+            if (loopStart.isValid()) {
+                const mixxx::audio::FramePos loopEnd = loopStart + pLoopCue->getLengthFrames();
+                if (loopEnd.isValid() && loopStart <= loopEnd) {
```

Comment:
loopEnd will never be invalid, because getLengthFrames() always returns a valid value and 0 for an invalid endPosition (= cue point).

Moreover does an empty loop make any sense? I guess not.

Reply:
f176928162f096ba60043e71f4793cb17db97d31